model-engine/model_engine_server/inference/forwarding/http_forwarder.py

```diff
@@ -70,11 +71,26 @@ def load_streaming_forwarder():
 @app.post("/predict")
 def predict(
     request: EndpointPredictV1Request,
+    background_tasks: BackgroundTasks,
     forwarder=Depends(load_forwarder),
     limiter=Depends(get_concurrency_limiter),
 ):
     with limiter:
-        return forwarder(request.dict())
+        try:
+            payload = request.dict()
+            response = forwarder(payload)
+            print(response)
+            if isinstance(response, JSONResponse):
+                loaded_response = json.loads(response.body)
```

Comment:
why not offload this to the `post_inference_hooks_handler.handle` method?

Reply:
ah that does make sense, let me try that

Reply:
hm so actually this needs to occur before `post_inference_hooks_handler.handle` since it takes a `dict` as the `response` and this logic converts the response to a `dict`